### DISCUSSION TITLE
درخواست آموزش قدم به قدم اجرای نسخه پایتون روی ویندوز

### DISCUSSION THREAD

**Discussion by HosseinR8** [MortezaBashsiz/CFScanner#499] [General]
سلام...لطفا آموزش قدم به قدم اجرای نسخه پایتون روی ویندوز رو بذارید...من هرکاری کردم نتونستم اجرا کنم...ممنون

**Comment by tempookian** [comment]
> سلام...لطفا آموزش قدم به قدم اجرای نسخه پایتون روی ویندوز رو بذارید...من هرکاری کردم نتونستم اجرا کنم...ممنون

سلام. لطفا دقیقا بگید چه اروری گرفتید. و مشکل کجاست تا راهنمایی کنیم و در آموزش‌های آتی هم لحاظ شه

**Reply by HosseinR8** [reply]
سلام...ممنون از پاسخگویی شما...
مراحل رو قدم به قدم میگم با خطایی که داد...
من نسخه 3.11 پایتون رو دانلود و روی ویندوز 11 نصب کردم...اسکنر هم دانلود کردم .....وارد پوشه پایتون اسکنر شدم ...کلیک راست و پاور شل رو اجرا کردم...اون سه تا لایبرری رو با دستور py -m pip install requests pysocks rich نصب کردم.....بعد از اون pip install cfscanner --upgrade رو اجرا کردم....در مرحله بعد cfscanner.py  رو که اجرا میکنم خطا میده...
cfscanner.py : The term 'cfscanner.py' is not recognized as the name of a cmdlet, function, script file, or operable
program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ cfscanner.py
+ ~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (cfscanner.py:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    لطفا بگید اشتباه من کجاست؟...
    یه سوال دیگه....در حال حاضر کامل ترین و دقیق ترین نسخه اسکنر کدومه؟نسخه ویندوز...پایتون یا...؟و اینکه روی ویندوز نسخه ویندوز رو اجرا کنیم بهتره یا نسخه پایتون و ..؟

**Reply by tempookian** [reply]
لطفا از این طریق نصب کنید
```bash
pip install cfscanner
```
یا
```bash
python -m pip install cfscanner
```


بعد از نصب برای اجرا:
```bash
cfscanner
```
یا
```bash
python -m cfscanner
```

**Reply by HosseinR8** [reply]
ممنون...python -m cfscanner اجرا شد ولی با این کد مستقیم میره برای اسکن کل ای پی ها و هیچ سوالی نمیکنه(در مورد تعداد ایپی که نیاز داریم-پینگ-سرعت دانلود و اپلود و...! و اینکه آی پی ورژن 4 میخوایم یا IPV6....اینارو چطوری مشخص کنیم براش؟کاش یه آموزش تصویری توی یوتیوب میذاشتین...
به سوالم هم اگه پاسخ بدین ممنونم ........در حال حاضر کامل ترین و دقیق ترین و سریع ترین نسخه اسکنر کدومه(منظورم کدوم نسبت به بقیه امکانات بیشتری داره و دقیق تر و سریع تر عمل میکنه)؟نسخه ویندوز...پایتون یا...؟و اینکه روی ویندوز نسخه ویندوز رو اجرا کنیم بهتره یا نسخه پایتون و...؟

**Comment by tempookian** [comment]
یه آموزش ابتدایی از امکانات رو [این‌جا](https://github.com/MortezaBashsiz/CFScanner/discussions/505) گذاشتم. 

در رابطه با آیپی v4 و v6 هم باید بگم که لیستی که به صورت پیش‌فرض اسکن می‌شه صرفا شامل آیپی‌های v4 کلاودفلره. اگر مایل بودید آیپی‌های v6 رو اسکن کنید می‌تونید لیستی ازشون داشته باشید و به صورت ورودی به اسکنر بدید. فرض کنید من فایلی دارم به نام ipsv6.txt که هر خط اون یک تک‌آیپی یا یک سابنت v6ئه. در این صورت می‌تونم به این صورت برنامه رو اجرا کنم:
```bash
python -m cfscanner -t 8 -r 50 -DS 450 -US 50 --shuffle-subnets -s ipsv6.txt
```

دقت کنید که این اسکنر به صورت interactive طراحی نشده و قصدی هم در این جهت فعلا وجود نداره. این به این معنیه که کلیه گزینه‌ها رو بایستی توی خط دستورتون وارد کنید. 

در مورد کامل‌ترین دقیق‌ترین هم باید بگم خیلی سلیقه‌ایه. بهتره که خودتون امتحان کنید و با هر کدوم راحت‌تر بودید کار کنید. اگر مشکل، ایراد یا پیشنهادی هم داشتید حتما به عنوان issue قرار بدید تا بررسی و اعمال بشه
